ibm/service/cis/data_source_ibm_cis_mtls.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	cisMtlsOutput                  = "mtls_certificates"
+	cisMtlsCertID                  = "cert_id"
+	cisMtlsCertName                = "cert_name"
+	cisMtlsCertFingerprint         = "cert_fingerprint"
+	cisMtlsCertAssociatedHostnames = "cert_associated_hostnames"
+	cisMtlsCertCreatedAt           = "cert_created_at"
+	cisMtlsCertUpdatedAt           = "cert_updated_at"
+	cisMtlsCertExpiresOn           = "cert_expires_on"
+)
+
+func DataSourceIBMCISMtls() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataIBMCISMtlsRead,
+
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			cisMtlsOutput: {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Container for response information.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						cisMtlsCertID: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate ID",
+						},
+						cisMtlsCertName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate name",
+						},
+						cisMtlsCertFingerprint: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate Fingerprint",
+						},
+						cisMtlsCertAssociatedHostnames: {
+							Type:        schema.TypeList,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Computed:    true,
+							Description: "Certificate Associated Hostnames",
+						},
+						cisMtlsCertCreatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate Created At",
+						},
+						cisMtlsCertUpdatedAt: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate Updated At",
+						},
+						cisMtlsCertExpiresOn: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Certificate Expires On",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataIBMCISMtlsRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).CisMtlsSession()
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("[ERROR] Error while getting the CisMtlsSession() %s %v", err, sess))
+	}
+
+	zoneID, crn, _ := flex.ConvertTftoCisTwoVar(d.Id())
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewListAccessCertificatesOptions(zoneID)
+
+	result, resp, err := sess.ListAccessCertificates(opt)
```

Comment:
Here we don't need any pagination..If the list has more elements ..does ListAccessCertificates provides all certificates at once?.. Generally for other services we need to have pagination to get complete list